assets/js/state/sagas/softwareUpdates.test.js

```diff
@@ -0,0 +1,88 @@
+import { faker } from '@faker-js/faker';
+
+import { recordSaga } from '@lib/test-utils';
+
+import { networkClient } from '@lib/network';
+import MockAdapter from 'axios-mock-adapter';
+
+import {
+  startLoadingSoftwareUpdates,
+  setSoftwareUpdates,
+  setSoftwareUpdatesErrors,
+  setEmptySoftwareUpdates,
+} from '@state/softwareUpdates';
+
+import { fetchSoftwareUpdates } from './softwareUpdates';
+
+describe('Software Updates saga', () => {
+  describe('Fetching Software Updates', () => {
+    it('should successfully fetch software updates', async () => {
+      const axiosMock = new MockAdapter(networkClient);
+      const hostId = faker.string.uuid();
+      const successfulResponse = [
+        {
+          date: '2023-05-18',
+          advisory_name: 'SUSE-15-SP4-2023-2245',
+          advisory_type: 'Bug Fix Advisory',
+          advisory_status: 'stable',
+          id: 2192,
+          advisory_synopsis: 'Recommended update for libzypp, zypper',
+          update_date: '2023-05-18',
+        },
+        {
+          from_epoch: ' ',
+          to_release: '150400.7.60.2',
+          name: 'openssl-1_1',
+          from_release: '150400.7.25.1',
+          to_epoch: ' ',
+          arch: 'x86_64',
+          to_package_id: 37454,
+          from_version: '1.1.1l',
+          to_version: '1.1.1l',
+          from_arch: 'x86_64',
+          to_arch: 'x86_64',
+        },
+      ];
+
+      axiosMock
+        .onGet(`/api/v1/hosts/${hostId}/software_updates`)
+        .reply(200, successfulResponse);
+
+      const dispatched = await recordSaga(fetchSoftwareUpdates, {
+        payload: { hostId },
+      });
+
+      expect(dispatched).toEqual([
+        startLoadingSoftwareUpdates(),
+        setSoftwareUpdates({ hostId, softwareUpdates: successfulResponse }),
+      ]);
+    });
+
+    it('should empty software updates settings on failed fetching', async () => {
+      const axiosMock = new MockAdapter(networkClient);
+      const hostId = faker.string.uuid();
+      const responses = [
+        { status: 404, body: { message: '404 Not found' } },
+        { status: 500, body: { message: 'java.lang.NullPointerException' } },
+      ];
+
+      /* eslint-disable no-await-in-loop */
+      /* eslint-disable no-restricted-syntax */
+      for (const { status, body } of responses) {
```

Comment:
If there's a way to refactor this loop, it would be welcome.
The main problem is looping with an `await`. I tried with `responses.forEach()` and also `Promise.all()` and neither worked. However ESLint does not like the presence of the `await` in the loop...

Reply:
Nice! Yeah `test.each()` solves the problem quite nicely! Merci!